src/hotspot/share/memory/allocation.cpp

```diff
@@ -111,7 +111,7 @@ void* ArenaObj::operator new(size_t size, Arena *arena) throw() {
 // AnyObj
 //
 
-void* AnyObj::operator new(size_t size, Arena *arena) throw() {
+void* AnyObj::operator new(size_t size, Arena *arena) {
```

Comment:
Please remind us what the issue is with `throw()` as this is a change to shared code and it looks very inconsistent to remove it only for this one definition of `operator new`.

Reply:
Hi David, this throw was removed for this method's declaration in the corresponding hpp file in an earlier commit, but the Committer forgot to remove the throw() from the definition as well (the error can be viewed in the GHA for the earlier versions of this PR). I believe gcc only errors on this in C++17 mode (for some weird reason) and not our current C++14 mode, which is why it gets a pass on other platforms

Reply:
Thanks for that.